hnn_core/tests/test_gui.py

```diff
@@ -422,6 +422,114 @@ def test_gui_adaptive_spectrogram():
     plt.close('all')
 
 
+def test_gui_layer2_dipole():
+    gui = HNNGUI()
+    gui.compose()
+    gui.params['N_pyr_x'] = 3
+    gui.params['N_pyr_y'] = 3
+
+    gui.run_button.click()
+    figid = 1
+    figname = f'Figure {figid}'
+    axname = 'ax1'
+    gui._simulate_viz_action("edit_figure", figname, axname, 'default',
+                             'layer2 dipole', {}, 'clear')
+    gui._simulate_viz_action("edit_figure", figname, axname, 'default',
+                             'layer2 dipole', {}, 'plot')
+    # Check if data is plotted on the axes
+    assert len(gui.viz_manager.figs[figid].axes) == 2
+    # Check default figs have data on their axis
+    assert gui.viz_manager.figs[figid].axes[0].has_data()
+    assert gui.viz_manager.figs[figid].axes[1].has_data()
+    plt.close('all')
+
+
+def test_gui_layer5_dipole():
+    gui = HNNGUI()
+    gui.compose()
+    gui.params['N_pyr_x'] = 3
+    gui.params['N_pyr_y'] = 3
+
+    gui.run_button.click()
+    figid = 1
+    figname = f'Figure {figid}'
+    axname = 'ax1'
+    gui._simulate_viz_action("edit_figure", figname, axname, 'default',
+                             'layer5 dipole', {}, 'clear')
+    gui._simulate_viz_action("edit_figure", figname, axname, 'default',
+                             'layer5 dipole', {}, 'plot')
+    # Check if data is plotted on the axes
+    assert len(gui.viz_manager.figs[figid].axes) == 2
+    # Check default figs have data on their axis
+    assert gui.viz_manager.figs[figid].axes[0].has_data()
+    assert gui.viz_manager.figs[figid].axes[1].has_data()
+    plt.close('all')
+
+
+def test_gui_spikes():
+    gui = HNNGUI()
+    gui.compose()
+    gui.params['N_pyr_x'] = 3
+    gui.params['N_pyr_y'] = 3
+
+    gui.run_button.click()
+    figid = 1
+    figname = f'Figure {figid}'
+    axname = 'ax1'
+    gui._simulate_viz_action("edit_figure", figname, axname, 'default',
+                             'spikes', {}, 'clear')
+    gui._simulate_viz_action("edit_figure", figname, axname, 'default',
+                             'spikes', {}, 'plot')
+    # Check if data is plotted on the axes
+    assert len(gui.viz_manager.figs[figid].axes) == 2
+    # Check default figs have data on their axis
+    assert gui.viz_manager.figs[figid].axes[0].has_data()
+    assert gui.viz_manager.figs[figid].axes[1].has_data()
+    plt.close('all')
+
+
+def test_gui_PSD():
```

Comment:
function names should all be lowercase [as per pep8 guide](https://peps.python.org/pep-0008/#function-and-variable-names)